### PR TITLE
feat: add test case for mongodb-3.x-plugin

### PR DESCRIPTION
**Pull request by scolia**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

### New feature or improvement

add test case for mongodb-3.x-plugin

### CODE REVIEW

File: test/plugin/scenarios/mongodb-3.x-scenario/bin/startup.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#!/bin/sh
+
+home="$(cd "$(dirname $0)"; pwd)"
+
+java -jar ${agent_opts}  "-Dskywalking.agent.service_name=mongodb-3.x-scenario" ${home}/../libs/mongodb-3.x-scenario.jar &
```
Comment (dmsolr) [inline test/plugin/scenarios/mongodb-3.x-scenario/bin/startup.sh]: We don't need to configure this.

File: test/plugin/scenarios/mongodb-3.x-scenario/config/expectedData.yaml [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  heartbeat: []
+
+segmentItems:
+- applicationCode: mongodb-3.x-scenario
+  segmentSize: 2
+  segments:
+  - segmentId: not null
+    spans:
+    - operationName: /case/healthCheck
+      operationId: 0
```
Comment (dmsolr) [inline test/plugin/scenarios/mongodb-3.x-scenario/config/expectedData.yaml]: health-check will be called more one when it misses a failure. So set `segmentSize` as `ge 2` and remove the segment of health-check.

File: test/plugin/scenarios/mongodb-3.x-scenario/src/main/java/org/apache/skywalking/apm/testcase/mongodb/controller/CaseController.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    @Value(value = "${mongodb.host}")
+    private String host;
+
+    @Value(value = "${mongodb.port:27017}")
+    private Integer port;
+
+    @GetMapping("/healthCheck")
+    public String health() {
+        return "success";
+    }
```
Comment (dmsolr) [inline test/plugin/scenarios/mongodb-3.x-scenario/src/main/java/org/apache/skywalking/apm/testcase/mongodb/controller/CaseController.java]: In fact, we must ensure the MongoDB server is available. so we need to connect to MongoDB server.

File: test/plugin/scenarios/mongodb-3.x-scenario/pom.xml [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+            <properties>
+                <test.framework.version>3.3.0</test.framework.version>
+            </properties>
+        </profile>
+        <profile>
+            <id>mongodb-3.x-scenario-3.4.0</id>
+            <properties>
+                <test.framework.version>3.4.0</test.framework.version>
+            </properties>
+        </profile>
```
Comment (dmsolr) [inline test/plugin/scenarios/mongodb-3.x-scenario/pom.xml]: we don't need these anymore. Please refer to https://github.com/apache/skywalking/blob/master/test/plugin/scenarios/canal-scenario/pom.xml

File: test/plugin/scenarios/mongodb-3.x-scenario/src/main/resources/log4j2.xml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        <logger name="com.a.eye.skywalking.ui" level="debug" additivity="false">
+            <AppenderRef ref="Console"/>
+        </logger>
+        <!--<logger name="org.skywalking.apm.dependencies" level="off">-->
+        <!--</logger>-->
+        <Root level="debug">
+            <AppenderRef ref="Console"/>
+        </Root>
+    </Loggers>
+</Configuration>
```
Comment (dmsolr) [inline test/plugin/scenarios/mongodb-3.x-scenario/src/main/resources/log4j2.xml]: Please refer to https://github.com/apache/skywalking/blob/master/test/plugin/scenarios/canal-scenario/src/main/resources/log4j2.xml

**Comment by wu-sheng** [comment]
@dmsolr I just removed the milestone, as I didn't understand the process.

File: Jenkinsfile-Agent-Test-3 [hunk 6]
```diff
@@ -75,6 +75,11 @@ pipeline {
                                 sh 'bash test/plugin/run.sh --build_id=wl3_${BUILD_ID} canal-scenario'
                             }
                         }
+                        stage('mongodb 3.0-3.11.1 (17)') {
+                            steps {
+                                sh 'bash test/plugin/run.sh --build_id=${BUILD_ID} mongodb-3.x-scenario'
```
Comment (wu-sheng) [inline Jenkinsfile-Agent-Test-3]: `--build_id=wl3_${BUILD_ID}` should replace --build_id=${BUILD_ID}

File: test/plugin/scenarios/mongodb-3.x-scenario/support-version.list [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+3.4.0
+3.4.1
+3.4.2
+3.4.3
+3.5.0
+3.7.0
```
Comment (wu-sheng) [inline test/plugin/scenarios/mongodb-3.x-scenario/support-version.list]: If the existing plugin didn't support 3.7+, then why all these versions are here?

**Comment by wu-sheng** [comment]
SOFARPC tests merged, please add yours into group 1 and resolve the conflicts.

**Comment by wu-sheng** [comment]
Welcome to be our new contributor.

**Pull request merged**